src/Compilers/Core/MSBuildTask/Csi.cs

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    public class Csi : InteractiveCompiler
+    {
+        #region Properties - Please keep these alphabetized.
+
+        // These are the parameters specific to Csi.
+        // The ones shared between Csi and Vbi are defined in InteractiveCompiler.cs, which is the base class.
+
+        public bool CheckForOverflowUnderflow
+        {
+            set
+            {
+                _store[nameof(CheckForOverflowUnderflow)] = value;
+            }
+
+            get
+            {
+                return _store.GetOrDefault(nameof(CheckForOverflowUnderflow), false);
+            }
+        }
+
+        #endregion
+
+        #region Tool Members
+        /// <summary>
+        /// Return the name of the tool to execute.
+        /// </summary>
+        protected override string ToolName
+        {
+            get
+            {
+                return "Csi.exe";
```

Comment:
I don't believe so, however this follows the same casing used in the Csc and Vbc tasks.